workshops/models.py

```diff
@@ -228,9 +228,15 @@ def get_by_ident(ident):
         The real indicator is the 'published' flag.
         '''
 
+        # match event ID
         if re.match(r'^\d+$', ident):
             return Event.objects.get(id=ident)
 
+        # match event slug in format YYYY-MM-****
+        if re.match(r'^\d{4}-\d{2}-.+$', ident):
```

Comment:
I think we can assume YYYY-MM-something, so there are just two cases: 
the numeric object ID, and something that looks like \d{4}-\d{2}-.+